File: app/src/main/kotlin/de/anubi1000/turnierverwaltung/database/model/Team.kt
package de.anubi1000.turnierverwaltung.database.model

import io.realm.kotlin.ext.backlinks
import io.realm.kotlin.ext.realmListOf
import io.realm.kotlin.query.RealmResults
import io.realm.kotlin.types.RealmList
import io.realm.kotlin.types.RealmObject
import io.realm.kotlin.types.annotations.PersistedName
import io.realm.kotlin.types.annotations.PrimaryKey
import org.mongodb.kbson.ObjectId

class Team(
    @PrimaryKey
    @PersistedName("_id")
    var id: ObjectId = ObjectId(),
    var name: String = "",
    var startNumber: Int = 0,
    var members: RealmList<Participant> = realmListOf(),
    var participatingDisciplines: RealmList<TeamDiscipline> = realmListOf(),
) : RealmObject {
    constructor() : this(id = ObjectId())

    val tournament: RealmResults<Tournament> by backlinks(Tournament::teams)
}

File: Server/Database/Model/Team.cs
using System.ComponentModel.DataAnnotations;
using Turnierverwaltung.Server.Utils;

namespace Turnierverwaltung.Server.Database.Model;

public sealed class Team
{
    // Common properties
    public int Id { get; init; }

    [MaxLength(Constants.MaxNameLength)]
    public string Name { get; set; } = "";

    public int StartNumber { get; set; }

    // Tournament
    public int TournamentId { get; init; }
    public Tournament Tournament { get; init; } = null!;

    // Members
    public ICollection<Participant> Members { get; set; } = new List<Participant>();

    // ParticipatingDisciplines
    public ICollection<TeamDiscipline> ParticipatingDisciplines { get; set; } = new List<TeamDiscipline>();
}
